### PR TITLE
Quick build fix for ScalNet

### PR DESCRIPTION
**Pull request by huitseeker**
- reimplement `ReshapePreProcessor`
- update `WeightInit.VI` -> `WeightInit.XAVIER_UNIFORM`
- allows optional building with sbt (using sbt-pomreader)
- runs defined examples as an integration test

### CODE REVIEW

**Comment by huitseeker** [comment]
ping @turambar

File: pom.xml [hunk 1]
```diff
@@ -55,8 +55,8 @@
     <scala.version>2.11.8</scala.version>
     <nd4j.version>0.7.3-SNAPSHOT</nd4j.version>
     <dl4j.version>0.7.3-SNAPSHOT</dl4j.version>
```
Comment (agibsonccc) [inline pom.xml]: Update versions?

Comment (saudet) [inline pom.xml]: We'll need to make the release first

**Comment by saudet** [comment]
If ReshapePreProcessor was removed from Deeplearing4j, shouldn't we remove it in ScalNet as well?

**Comment by huitseeker** [comment]
@saudet The reshaping is fundamental to ScalNet. I'll let @turambar argue:
screenshot-dave

**Comment by turambar** [comment]
Reasons to support generic reshaping:
* ScalNet is meant to be an implementation of the [Keras spec](https://blog.keras.io/introducing-keras-2.html?t=1) and Keras has a generic [Reshape](https://keras.io/layers/core/#reshape) layer (as well as [Permute](https://keras.io/layers/core/#permute) and a few others). Thus, we need to support that functionality, even if we don't provide a DL4J generic reshaping InputPreProcessor.
* We need expanded reshaping functionality in order to implement a [TimeDistributed](https://github.com/deeplearning4j/deeplearning4j/issues/2865) layer anyway.

**Comment by turambar** [comment]
But if I'm playing devil's advocate, as far as I know, the number of actual use cases for the generic Reshape layer in Keras are pretty limited. Uses of Flatten are pretty widespread (any convolutional neural network), but the TimeDistributed wrapper knocks out a lot of the possible reshaping use cases.

**Comment by huitseeker** [comment]
Quoting @AlexDBlack on another channel: 
screenshot-alex

So, IIUC, checks for a first version include:
- adding the required output ordering as an argument
- making sure the prior layer has the same ordering as the required output, by making sure `input.ordering()` is what's expected

Anything else ?

**Comment by huitseeker** [comment]
Can I leave an issue for the two missing checks above and merge this for release ? This PR adds a private class, so it is not a commitment to supporting generic reshapes yet.

**Comment by AlexDBlack** [comment]
I'm totally fine with a generic reshaping layer being added to ScalNet. But: that particular reshaping layer implementation needs some work.

Tests: At a minimum: we'll want to test c and f orders on both the forward and backward pass - to make sure that the input order doesn't make any difference to the actual output. Tests for edge cases - like 1s in the shape - wouldn't hurt either. These tests are not a bad place to start for that:
https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-core/src/test/java/org/deeplearning4j/nn/conf/preprocessor/TestPreProcessors.java

Gradient checks would be good, but if  the standard unit tests are thorough enough we should be OK.

As for ensuring those tests pass: We need to decide on a convention for the input order - either C or F - and first duplicate (with order) if it's the wrong order input (and ideally, also duplicate to if Shape. strideDescendingCAscendingF returns false). I'd go with C as the default (better for CNNs and MLPs), *but allow people to customize this*. We use F order arrays for RNNs (and, RNN activation reshaping) for performance reasons (strides are bad with RNN + C order, which has a noticeable performance impact). I'm thinking we always do c->c or f->f reshaping based on the order arg (then it's just one configuration option, and matches how DL4J's preprocessors work).

**Comment by saudet** [comment]
Thanks @AlexDBlack for the feedback. For now though, would it be alright to merge this and get 0.8.0 out the door? Or is there something that we should absolutely do before that happens?

**Comment by AlexDBlack** [comment]
@saudet Unless someone has the bandwidth to actually look at this (and, write proper tests) it might be better to just get it out, log it as a known issue, and fix it later? That's not ideal, but neither is the delay...

**Comment by turambar** [comment]
Concur with @AlexDBlack.
